exporter/prometheusexporter/accumulator_test.go

```diff
@@ -26,46 +26,173 @@ import (
+		dp.Attributes().PutStr("label_1", "1")
+		dp.Attributes().PutStr("label_2", "2")
+		dp.SetTimestamp(pcommon.NewTimestampFromTime(ts))
+		dp.SetStartTimestamp(pcommon.NewTimestampFromTime(startTs))
 	}
 
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			resourceMetrics := pmetric.NewResourceMetrics()
-			ilm := resourceMetrics.ScopeMetrics().AppendEmpty()
-			ilm.Scope().SetName("test")
-			tt.fillMetric(time.Now(), ilm.Metrics().AppendEmpty())
+	startTs1 := time.Now().Add(-6 * time.Second)
+	startTs2 := time.Now().Add(-5 * time.Second)
+	startTs3 := time.Now()
+	ts1 := time.Now().Add(-4 * time.Second)
+	ts2 := time.Now().Add(-3 * time.Second)
+	ts3 := time.Now().Add(-2 * time.Second)
+	ts4 := time.Now().Add(-1 * time.Second)
+	ts5 := time.Now().Add(1 * time.Second)
+
+	a := newAccumulator(zap.NewNop(), 1*time.Hour).(*lastValueAccumulator)
+
+	resourceMetrics1 := pmetric.NewResourceMetrics()
+	ilm1 := resourceMetrics1.ScopeMetrics().AppendEmpty()
+	ilm1.Scope().SetName("test")
+	// counts is one more than explicit bounds to account for the one implicit count/bucket for <=inf
+	appendHistogram(startTs2, ts3, 5, 2.5, []uint64{1, 3, 1, 0, 0}, []float64{0.1, 0.5, 1, 10}, ilm1.Metrics())
+	appendHistogram(startTs2, ts2, 4, 8.3, []uint64{1, 1, 2, 0, 0}, []float64{0.1, 0.5, 1, 10}, ilm1.Metrics())
+
+	m3 := ilm1.Metrics().At(0).Histogram().DataPoints().At(0)
+	m2 := ilm1.Metrics().At(1).Histogram().DataPoints().At(0)
+	signature := timeseriesSignature(ilm1.Scope().Name(), ilm1.Metrics().At(0), m2.Attributes(), pcommon.NewMap())
+
+	// different buckets from the past
+	resourceMetrics2 := pmetric.NewResourceMetrics()
+	ilm2 := resourceMetrics2.ScopeMetrics().AppendEmpty()
+	ilm2.Scope().SetName("test")
+	appendHistogram(startTs2, ts1, 7, 5, []uint64{3, 1, 1, 0, 0}, []float64{0.1, 0.2, 1, 10}, ilm2.Metrics())
+
+	// add extra buckets
+	resourceMetrics3 := pmetric.NewResourceMetrics()
+	ilm3 := resourceMetrics3.ScopeMetrics().AppendEmpty()
+	ilm3.Scope().SetName("test")
+	appendHistogram(startTs2, ts4, 7, 5, []uint64{3, 1, 1, 0, 0, 0}, []float64{0.1, 0.2, 1, 10, 15}, ilm3.Metrics())
+
+	m4 := ilm3.Metrics().At(0).Histogram().DataPoints().At(0)
+
+	// misaligned start timestamp, drop
+	resourceMetrics4 := pmetric.NewResourceMetrics()
+	ilm4 := resourceMetrics4.ScopeMetrics().AppendEmpty()
+	ilm4.Scope().SetName("test")
+	appendHistogram(startTs1, ts5, 4, 8.3, []uint64{1, 1, 2, 0, 0}, []float64{0.1, 0.5, 1, 10}, ilm4.Metrics())
+	appendHistogram(ts3, ts5, 4, 8.3, []uint64{1, 1, 2, 0, 0}, []float64{0.1, 0.5, 1, 10}, ilm4.Metrics())
+
+	// misaligned start timestamp, treat as restart
+	resourceMetrics5 := pmetric.NewResourceMetrics()
+	ilm5 := resourceMetrics5.ScopeMetrics().AppendEmpty()
+	ilm5.Scope().SetName("test")
+	appendHistogram(startTs3, ts5, 4, 8.3, []uint64{1, 1, 2, 0, 0}, []float64{0.1, 0.5, 1, 10}, ilm5.Metrics())
+	m5 := ilm5.Metrics().At(0).Histogram().DataPoints().At(0)
+
+	t.Run("Accumulate", func(t *testing.T) {
```

Comment:
Are these independent tests or does one test affect the operation of the next?  Can they be separated out?  It's somewhat hard to follow what's being done here.

Reply:
Agreed that having all the test setup in a single place makes it hard to follow. These were not independent tests, but I just did a re-write to have independent setups for each. Hopefully the tests are easier to follow now.